tests/test_cli.py

```diff
@@ -305,6 +305,16 @@ def test_run_with_user_yamllint_config_file_in_env(self):
                 cli.run((os.path.join(self.wd, 'a.yaml'), ))
             self.assertEqual(ctx.returncode, 1)
 
+    def test_run_with_locale(self):
+        loc = locale.getlocale(locale.LC_ALL)
+        self.assertNotEqual(loc, 'en_US.UTF-8')
```

Comment:
My understanding of this line is that you want to make sure `cli.run(('-d', 'locale: en_US.UTF-8',` actually changes something. But for our American friends, it may result in crashes when trying to package yamllint (if their system default is `en_US.UTF-8`)...

Instead, maybe we could drop this line, but repeat the test code with a second locale, e.g. `de_DE`?

Reply:
I'm not sure that is actually going to be a problem (because the system default is only used, when explicitly setting `locale: ""` and the default should be `C.UTF-8` as set in config.py), but that line is faulty anyways (`getlocale` returns a tuple instead of a string, see the `assertEqual` below).

Your suggestion to repeat the test is good, I guess we would be on the safe side. I choose `en_US.UTF-8` under the assumption that this locale should be installed in most systems. I guess changing back to `C.UTF-8` instead of `de_DE` for the 2nd test is better, because the tests would not depend on `de_DE` being available?

Reply:
So I played around a little bit - and that whole test-case is completely broken right now. `getlocale` just does not return expected values at all.

I'll probably just add a more implicit test here - by linting some basic yaml file with the key-ordering rule and a locale setting to see it having an effect.

Reply:
Interesting! OK, looking forward to see that.

Reply:
PS: For the record, I remember having packaging problems because `en_US.UTF-8` or `C.UTF-8` were not available on some systems (CentOS 6 or 7 I think).

Reply:
So I changed the test to actually run some key-ordering rules against files and look at the return code to see whether the `locale` option had any effect.

Those tests are skipped if the `en_US.UTF-8` locale is not present on the system - same for the key-ordering locale tests.

But since `C.UTF-8` is set as the default, I did not do any checks here. Tests will fail (probably everywhere?) when `C.UTF-8` is not available.

Reply:
That seems reasonable. I hope there aren't any systems where `C.UTF-8` is not available.